Cut 2.5.0-rc.0

brancz opened the pull request:
Pretty impressive list for a 6 week time window. Please check and make sure the entries I added make sense and that all work is reflected :slightly_smiling_face: .

@brian-brazil @juliusv @krasi-georgiev @simonpasquier @gouthamve @cstyan @csmarchbanks

mdlayher commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+## 2.5.0-rc.0 / 2018-10-23
+
+Starting with the s390x platform for Linux is officially supported.
```
* "Starting with this release"?

SuperQ replied:
We tend to not list changes like build stuff in our release notes.

brancz replied:
@SuperQ agreed, but supporting an additional platform seems like it belongs in the changelog. I'm actually interested what we mean by this (I took the "support" statement right from the PR that merged this). For this and other platforms that are not amd64 we should probably clearer on what support means as we only run tests on amd64 and arguably all of us probably don't run Prometheus on anything other (maybe some arm devices). So "supportability" becomes somewhat undefined.

gouthamve commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+
+Starting with the s390x platform for Linux is officially supported.
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
```
This made it into `2.4.0`

gouthamve commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+Starting with the s390x platform for Linux is officially supported.
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
```
Same for this.

gouthamve commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [ENHANCEMENT] OpenStack SD: Support discovering instances from all projects.
+* [ENHANCEMENT] OpenStack SD: Discover all interfaces.
+* [ENHANCEMENT] OpenStack SD: Support `tls_config` for the used HTTP client.
+* [ENHANCEMENT] EC2 SD: Expose owner of an instance as meta label.
```
Made it into 2.4.0

gouthamve commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [ENHANCEMENT] Web UI: Avoid browser spell-checking in expression field.
+* [ENHANCEMENT] Web UI: Add scrape duration and last evaluation time in targets and rules pages.
+* [ENHANCEMENT] Web UI: Improve rule view by wrapping lines.
+* [BUGFIX] Fix histograms on 32 bit platforms.
```
This never made it into a release and can be removed

brian-brazil commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [ENHANCEMENT] Make maximum number of concurrent remote reads configurable via `--storage.remote.read-concurrent-limit` flag.
+* [ENHANCEMENT] Remote Read: Add `prometheus_api_remote_read_queries` metric tracking currently executed or waiting remote read queries.
+* [ENHANCEMENT] Remote Read: Reduced memory usage.
+* [ENHANCEMENT] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify as well as by section in the respective configuration.
```
This is a CHANGE

simonpasquier commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [BUGFIX] Fix histograms on 32 bit platforms.
+* [BUGFIX] Change max/min over_time to handle NaNs properly.
+* [BUGFIX] Discovery: Properly unregister source when targets are empty.
+* [BUGFIX] Fix asset paths for Windows platforms.
```
Was already released in v2.4.1

simonpasquier commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [BUGFIX] Change max/min over_time to handle NaNs properly.
+* [BUGFIX] Discovery: Properly unregister source when targets are empty.
+* [BUGFIX] Fix asset paths for Windows platforms.
+* [BUGFIX] Don't remove final dot from rooted DNS names.
```
This one is still controversial, see #4766 .

grobie replied:
I'm in favor of reverting this change, see my comment in #4766. If you decide against it, this is a significant `CHANGE` for any dns discovery user and should be highlighted as such.

brian-brazil commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
+* [FEATURE] Add OpenMetrics support (EXPERIMENTAL).
```
For scraping.

simonpasquier commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [ENHANCEMENT] Web UI: Improve rule view by wrapping lines.
+* [BUGFIX] Fix histograms on 32 bit platforms.
+* [BUGFIX] Change max/min over_time to handle NaNs properly.
+* [BUGFIX] Discovery: Properly unregister source when targets are empty.
```
This bug fix was in the file SD adapter code so it probably doesn't warrant an entry.

brancz commented:
@gouthamve PR numbering we only started doing recently but happy to add it, will make sure to add it to https://github.com/prometheus/docs/pull/1169 as well then.

cstyan commented on CHANGELOG.md:
```diff
@@ -1,3 +1,36 @@
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
+* [FEATURE] Add OpenMetrics support (EXPERIMENTAL).
+* [FEATURE] Add unit testing for rules.
+* [ENHANCEMENT] Make maximum number of samples per query configurable via `--query.max-samples` flag.
```
:+1: 

(pr #4513 )

brancz commented:
I updated the PR to inline the respective pull request numbers and updated the release date to today's date. It seems we also want to have some consensus on https://github.com/prometheus/prometheus/pull/4766.

juliusv commented on CHANGELOG.md:
```diff
@@ -1,3 +1,32 @@
+## 2.5.0-rc.0 / 2018-10-24
+
+* [CHANGE] Marathon SD: Various changes to adapt to marathon 1.5+. #4499
```
```suggestion
* [CHANGE] Marathon SD: Various changes to adapt to Marathon 1.5+. #4499
```

juliusv commented on CHANGELOG.md:
```diff
@@ -1,3 +1,32 @@
+## 2.5.0-rc.0 / 2018-10-24
+
+* [CHANGE] Marathon SD: Various changes to adapt to marathon 1.5+. #4499
+* [CHANGE] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify as well as by section in the respective configuration. #4753
```
```suggestion
* [CHANGE] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify (Alertmanager SD) as well as by section in the respective configuration. #4753
```

juliusv commented:
At least from a breakage perspective (as outlined by @grobie this can be quite severe) I'd be for #4766, that is, reverting the DNS dot removal fix and going back to the behavior we had.

brian-brazil commented:
:+1:
